patterns/Prototype/Prototype.js

```diff
@@ -0,0 +1,20 @@
+// Prototype object
+const vehiclePrototype = {
+    init(make, model) {
+      this.make = make;
+      this.model = model;
+    },
+    getDetails() {
+      return `${this.make} ${this.model}`;
+    },
+  };
+  
+  // Create new instances using the prototype
+  const car1 = Object.create(vehiclePrototype);
+  car1.init('Toyota', 'Camry');
```

Comment:
One of the benefits of the prototype pattern is that repeated initialization of common objects is less cumbersome because you can reuse some initial values and only change the necessary ones (in contrast to instantiating a new class object, which typically always requires full initialization).

In your examples here, you've sidestepped this benefit, since you've initialized both objects from scratch. It may be preferable to show an example that demonstrates a more complex base object (with more parameters) and smaller changes in the copied objects.